### ISSUE TITLE
ServiceMonitor/rook-ceph-mgr uses incorrect selector

### ISSUE THREAD

**Issue by TheRealNoob** [rook/rook#12664]
Hello.  I am new to rook so some of my understanding may be askew from reality.  Thank you to the hard work of everyone :heart:.

**Bug report:**

The operator is deploying a ServiceMonitor whose selectors do not match the labels on the Service.  These objects look like:

```yaml
kind: Service
metadata:
  name: rook-ceph-mgr
  labels:
    app: rook-ceph-mgr
    rook_cluster: ceph
spec:
  selector:
    app: rook-ceph-mgr
    mgr_role: active
    rook_cluster: ceph
```

```yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: rook-ceph-mgr
spec:
  selector:
    matchLabels:
      app: rook-ceph-mgr
      mgr_role: active
      rook_cluster: ceph
```

**Expected behavior:**

Either the ServiceMonitor selector needs to change, or the labels on the Service.  **Note** that the labels on `Service/rook-ceph-mgr-dashboard` are identical to those `Service/rook-ceph-mgr`, so the ServiceMonitor would identify both Services if `mgr_role: active` was removed from the ServiceMonitor selector.  However at the moment (for me) both services point to the same pod(s), so in this case I don't think it makes much difference, but it would still be a non-ideal outcome and something to watch out for.

**How to reproduce it (minimal and precise):**

My helm values looks like

`helm install rook-ceph rook/rook-ceph-cluster -n ceph --version 1.11.6 -f values.yaml`

```yaml
operatorNamespace: rook

monitoring:
  enabled: true
  createPrometheusRules: true

cephClusterSpec:
  labels:
    monitoring:
      release: prometheus
```

**Comment by travisn** [comment]
What version of Rook do you have. There have been several fixes around the labels and service monitor recently, such as #12293. 

The dashboard and mgr services won't conflict since they are on different ports, so that should not be an issue. Does that answer the concern? If not, I must not be following the suggested issue.

**Comment by TheRealNoob** [comment]
hmmm you may be correct, I updated from 1.11.6 to 1.12.1 and it seems to be solved now.  Thank you!

**Comment by ExNG** [comment]
Can confirm this issue has been fixed since at least v1.12.1
